app/models/activity.rb

```diff
@@ -247,9 +247,19 @@ def descendants
     Activity.find_by_sql([sql, id])
   end
 
-  def total_spend_to_date
+  def total_spend(financial_quarter = nil)
     activity_ids = descendants.pluck(:id).append(id)
-    Transaction.where(parent_activity_id: activity_ids).sum(:value)
+    transactions = if financial_quarter
```

Comment:
It might be clearer to pull out the common `where` and remove the `else`, i.e.

```ruby
transactions = Transaction.where(parent_activity_id: activity_id)

if financial_quarter
  transactions = transactions.where(financial_year: ...)
end

transactions.sum(:value)
```